### PR TITLE
refactors radios to use proximity

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
refactors radios to use proximity

## Why It's Good For The Game
Overtime reduction, easier to understand code, WAY faster

## Testing
Compiled and ran, worked fine in game

## Changelog
NPFC

### CODE REVIEW

File: code/game/objects/items/devices/radio/radio_objects.dm [hunk 1]
```diff
@@ -116,6 +117,31 @@ GLOBAL_LIST_EMPTY(deadsay_radio_systems)
...
+	RegisterSignal(crosser, COMSIG_MOVABLE_MOVED, PROC_REF(is_crosser_still_listening))
+	RegisterSignal(crosser, COMSIG_PARENT_QDELETING, PROC_REF(remove_from_listener_list))
+
+/obj/item/radio/proc/is_crosser_still_listening(mob/crosser)
+	SIGNAL_HANDLER
+	var/still_listening = FALSE
+	for(var/obj/effect/abstract/proximity_checker/checker in get_turf(crosser))
+		if(checker.monitor.hasprox_receiver == src)
+			still_listening = TRUE
+			break
```
Comment (GDNgit) [inline code/game/objects/items/devices/radio/radio_objects.dm:136]:
```suggestion
	for(var/obj/effect/abstract/proximity_checker/checker in get_turf(crosser))
		if(checker.monitor.hasprox_receiver == src)
			still_listening = TRUE
			break
```
This is scuffed but I don't think we have a better way to do this. It also looks way worse than it actually is

File: code/game/objects/items/devices/radio/radio_objects.dm [hunk 2]
```diff
@@ -202,14 +228,25 @@ GLOBAL_LIST_EMPTY(deadsay_radio_systems)
...
+/obj/item/radio/proc/update_hear_range(new_hear_range)
+	canhear_range = new_hear_range
+	var/datum/component/proximity_monitor/our_prox_component = GetComponent(/datum/component/proximity_monitor)
+	if(canhear_range)
+		if(our_prox_component)
+			our_prox_component.set_radius(canhear_range)
+		else
+			AddComponent(/datum/component/proximity_monitor, canhear_range)
+	else
+		QDEL_NULL(our_prox_component)
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/radio_objects.dm:248]:
```suggestion
	if(!canhear_range)
		QDEL_NULL(our_prox_component)
		return

	if(our_prox_component)
		our_prox_component.set_radius(canhear_range)
	else
		AddComponent(/datum/component/proximity_monitor, canhear_range)
```

File: code/game/objects/items/devices/radio/radio_objects.dm [hunk 3]
```diff
@@ -797,3 +832,16 @@ GLOBAL_LIST_EMPTY(deadsay_radio_systems)
 		return
 
 	return M.say_dead(message)
+
+GLOBAL_DATUM_INIT(deadchat_radio, /obj/item/radio/dchat_radio_handler, new)
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/radio_objects.dm:836]:
```suggestion
GLOBAL_DATUM_INIT(deadchat_radio, /obj/item/radio/dchat_radio_handler, new())
```

File: code/game/objects/items/devices/radio/radio_objects.dm [hunk 4]
```diff
@@ -797,3 +832,16 @@ GLOBAL_LIST_EMPTY(deadsay_radio_systems)
...
+
+/obj/item/radio/dchat_radio_handler
+	name = "Deadchat radio handler. Do not fuck with"
+	canhear_range = 0
+
+/obj/item/radio/dchat_radio_handler/Destroy()
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/radio_objects.dm:842]:
```suggestion
/obj/item/radio/dchat_radio_handler/Destroy(force)
```

**Review by Burzah — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]
LGTM

**Pull request merged**